<a href="https://colab.research.google.com/github/amnasyed1/datasci_5_statistics/blob/main/datasci_5_statistics.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## **Chi-Square Test**


In [ ]:
import pandas as pd
import scipy
from scipy.stats import chi2_contingency as chi2

In [11]:
# The dataset contains data from a survey where individuals who work in the tech industry completed about mental health
mhtech = pd.read_csv('https://raw.githubusercontent.com/amnasyed1/datasci_5_statistics/main/datasets/mental_health_in_tech_survey.csv')
mhtech

Timestamp  Age  Gender         Country state self_employed  \
0     2014-08-27 11:29:31   37  Female   United States    IL           NaN   
1     2014-08-27 11:29:37   44       M   United States    IN           NaN   
2     2014-08-27 11:29:44   32    Male          Canada   NaN           NaN   
3     2014-08-27 11:29:46   31    Male  United Kingdom   NaN           NaN   
4     2014-08-27 11:30:22   31    Male   United States    TX           NaN   
...                   ...  ...     ...             ...   ...           ...   
1254  2015-09-12 11:17:21   26    male  United Kingdom   NaN            No   
1255  2015-09-26 01:07:35   32    Male   United States    IL            No   
1256  2015-11-07 12:36:58   34    male   United States    CA            No   
1257  2015-11-30 21:25:06   46       f   United States    NC            No   
1258  2016-02-01 23:04:31   25    Male   United States    IL            No   

     family_history treatment work_interfere    no_employees  ...  \
0                No       Yes          Often            6-25  ...   
1                No        No         Rarely  More than 1000  ...   
2                No        No         Rarely            6-25  ...   
3               Yes       Yes          Often          26-100  ...   
4                No        No          Never         100-500  ...   
...             ...       ...            ...             ...  ...   
1254             No       Yes            NaN          26-100  ...   
1255            Yes       Yes          Often          26-100  ...   
1256            Yes       Yes      Sometimes  More than 1000  ...   
1257             No        No            NaN         100-500  ...   
1258            Yes       Yes      Sometimes          26-100  ...   

                   leave mental_health_consequence phys_health_consequence  \
0          Somewhat easy                        No                      No   
1             Don't know                     Maybe                      No   
2     Somewhat difficult                        No                      No   
3     Somewhat difficult                       Yes                     Yes   
4             Don't know                        No                      No   
...                  ...                       ...                     ...   
1254       Somewhat easy                        No                      No   
1255  Somewhat difficult                        No                      No   
1256  Somewhat difficult                       Yes                     Yes   
1257          Don't know                       Yes                      No   
1258          Don't know                     Maybe                      No   

         coworkers    supervisor mental_health_interview  \
0     Some of them           Yes                      No   
1               No            No                      No   
2              Yes           Yes                     Yes   
3     Some of them            No                   Maybe   
4     Some of them           Yes                     Yes   
...            ...           ...                     ...   
1254  Some of them  Some of them                      No   
1255  Some of them           Yes                      No   
1256            No            No                      No   
1257            No            No                      No   
1258  Some of them            No                      No   

     phys_health_interview mental_vs_physical obs_consequence comments  
0                    Maybe                Yes              No      NaN  
1                       No         Don't know              No      NaN  
2                      Yes                 No              No      NaN  
3                    Maybe                 No             Yes      NaN  
4                      Yes         Don't know              No      NaN  
...                    ...                ...             ...      ...  
1254                    No         Don't know              No      NaN  
1255    

## Hypothesis

Hypothesis: There is a correlation between working remote and it interfering with one's work if they have a mental health condition.

- Null Hypothesis (H0): There is no relationship or correlation between working remote and if the individual has a mental health condition, it interfering with their work.

- Alternative Hypothesis (H1): There is a correlation or relationship between working remote and for those who have a mental health condition, it interfering with their work.

In [4]:
mhtech.columns

Index(['Timestamp', 'Age', 'Gender', 'Country', 'state', 'self_employed',
       'family_history', 'treatment', 'work_interfere', 'no_employees',
       'remote_work', 'tech_company', 'benefits', 'care_options',
       'wellness_program', 'seek_help', 'anonymity', 'leave',
       'mental_health_consequence', 'phys_health_consequence', 'coworkers',
       'supervisor', 'mental_health_interview', 'phys_health_interview',
       'mental_vs_physical', 'obs_consequence', 'comments'],
      dtype='object')

In [5]:
mhtech['remote_work'].value_counts()

No     883
Yes    376
Name: remote_work, dtype: int64

In [6]:
mhtech['work_interfere'].value_counts()

Sometimes    465
Never        213
Rarely       173
Often        144
Name: work_interfere, dtype: int64

In [7]:
contingency_table = pd.crosstab(mhtech['remote_work'], mhtech['work_interfere'])
print(contingency_table)

work_interfere  Never  Often  Rarely  Sometimes
remote_work                                    
No                157     93     119        322
Yes                56     51      54        143


In [8]:
chi2, p, _, _ = chi2(contingency_table)
print(f"Chi2 value: {chi2}")
print(f"P-value: {p}")

Chi2 value: 3.4731660005348104
P-value: 0.3242591134502088


## Results

Significance Level = 0.05

P-value: 0.3242591134502088

Due to the p-value being greater than the significance level, we can reject the H1 because there is not enough evidence to reject the H0. The analysis shows that there is not a strong correlation between working remotely and one's mental health condition interfering with their work.

## Rational

I chose to examine the "remote_work" and "work_interfere" columns due to the trend and increasing number of jobs/companies, especially in the tech industry, having the option for employees to work remotely or requiring their employees to work remotely. Working remotely may be isolating, and for individuals with mental health conditions it may interfere with their work.

# **T-Test**

In [30]:
import pandas as pd
from scipy.stats import ttest_ind

In [31]:
brca = pd.read_csv('https://raw.githubusercontent.com/amnasyed1/datasci_5_statistics/main/datasets/BRCA.csv')
brca.head(10)

Patient_ID   Age  Gender  Protein1  Protein2  Protein3  Protein4  \
0  TCGA-D8-A1XD  36.0  FEMALE  0.080353   0.42638  0.547150  0.273680   
1  TCGA-EW-A1OX  43.0  FEMALE -0.420320   0.57807  0.614470 -0.031505   
2  TCGA-A8-A079  69.0  FEMALE  0.213980   1.31140 -0.327470 -0.234260   
3  TCGA-D8-A1XR  56.0  FEMALE  0.345090  -0.21147 -0.193040  0.124270   
4  TCGA-BH-A0BF  56.0  FEMALE  0.221550   1.90680  0.520450 -0.311990   
5  TCGA-AO-A1KQ  84.0    MALE -0.081872   1.72410 -0.057335  0.043025   
6  TCGA-D8-A73X  53.0  FEMALE -0.069535   1.41830 -0.361050  0.391580   
7  TCGA-A7-A426  50.0  FEMALE  0.672490   1.27900 -0.321070 -0.112390   
8  TCGA-EW-A1P5  77.0  FEMALE -0.151750  -0.66332  1.189400  0.217180   
9  TCGA-A8-A09A  40.0  FEMALE -0.565700   1.26680 -0.293460  0.193950   

  Tumour_Stage                       Histology ER status PR status  \
0          III   Infiltrating Ductal Carcinoma  Positive  Positive   
1           II              Mucinous Carcinoma  Positive  Positive   
2          III   Infiltrating Ductal Carcinoma  Positive  Positive   
3           II   Infiltrating Ductal Carcinoma  Positive  Positive   
4           II   Infiltrating Ductal Carcinoma  Positive  Positive   
5          III   Infiltrating Ductal Carcinoma  Positive  Positive   
6           II   Infiltrating Ductal Carcinoma  Positive  Positive   
7          III   Infiltrating Ductal Carcinoma  Positive  Positive   
8           II   Infiltrating Ductal Carcinoma  Positive  Positive   
9           II  Infiltrating Lobular Carcinoma  Positive  Positive   

  HER2 status                 Surgery_type Date_of_Surgery Date_of_Last_Visit  \
0    Negative  Modified Radical Mastectomy       15-Jan-17          19-Jun-17   
1    Negative                   Lumpectomy       26-Apr-17          09-Nov-18   
2    Negative                        Other       08-Sep-17          09-Jun-18   
3    Negative  Modified Radical Mastectomy       25-Jan-17          12-Jul-17   
4    Negative                        Other       06-May-17          27-Jun-19   
5    Negative  Modified Radical Mastectomy       18-Sep-17          15-Nov-21   
6    Negative            Simple Mastectomy       04-Feb-17          07-Feb-18   
7    Negative            Simple Mastectomy       16-May-17                NaN   
8    Negative  Modified Radical Mastectomy       28-Sep-17          28-Sep-18   
9    Positive                        Other       14-Feb-17          15-Dec-17   

  Patient_Status  
0          Alive  
1           Dead  
2          Alive  
3          Alive  
4           Dead  
5          Alive  
6          Alive  
7            NaN  
8          Alive  
9          Alive

## Hypothesis

Null Hypothesis (H0): There is no correlation between the expression levels of Protien4 and the Patient Status

Alternative Hypothesis (H1): There is a correlation between the expression levels of Protien4 and the Patient Status


In [34]:
brca.columns

Index(['Patient_ID', 'Age', 'Gender', 'Protein1', 'Protein2', 'Protein3',
       'Protein4', 'Tumour_Stage', 'Histology', 'ER status', 'PR status',
       'HER2 status', 'Surgery_type', 'Date_of_Surgery', 'Date_of_Last_Visit',
       'Patient_Status'],
      dtype='object')

In [37]:
dead_data = brca[brca['Patient_Status'] == 'Dead']['Protein4']
dead_data

1     -0.031505
4     -0.311990
11    -0.350700
18     0.601030
25     1.629900
         ...   
329   -0.517350
330   -0.354920
331    0.339680
332   -0.188480
333   -0.067848
Name: Protein4, Length: 66, dtype: float64

In [38]:
alive_data = brca[brca['Patient_Status'] == 'Alive']['Protein4']
alive_data

0      0.273680
2     -0.234260
3      0.124270
5      0.043025
6      0.391580
         ...   
322    1.044100
323    0.351920
324    0.261420
325    0.458740
326    0.721820
Name: Protein4, Length: 255, dtype: float64

In [40]:
# Assuming unequal variances with Welch's t-test # noqa
t_stat, p_val = ttest_ind(alive_data, dead_data)
t_stat, p_val = ttest_ind(alive_data, dead_data, equal_var=False)
print(f"T-statistic: {t_stat}")
print(f"P-value: {p_val}")

T-statistic: -1.4214132952738339
P-value: 0.1582830197703266


In [41]:
# Compute means for alive and dead data
alive_mean = alive_data.mean()
dead_mean = dead_data.mean()

print(f"Mean prevalence for southern states: {alive_mean}")
print(f"Mean prevalence for non-southern states: {dead_mean}")

Mean prevalence for southern states: -0.016361614666666666
Mean prevalence for non-southern states: 0.10607757575757576


## Results

Significance Level = 0.05
P-value: 0.1582830197703266


Due to the p-value being greater than the significance level, we can reject the H1 because there is not enough evidence to reject the H0. The test shows that there is not a strong correlation  between the patient's status (alive or expired) and the expression levels of Protien4.

## Rational

I chose to examine "Protien4" and "Patient_Status" columns to analyze if Protein4 expression levels have any type of correlations to a breast cancer patient's death.


# **ANOVA**

In [43]:
import statsmodels.api as sm
from statsmodels.formula.api import ols

In [44]:
asthma = pd.read_csv('https://raw.githubusercontent.com/amnasyed1/datasci_5_statistics/main/datasets/current-asthma-prevalence-by-county-2015_2020.csv')
asthma

COUNTY      YEARS            STRATA  AGE GROUP CURRENT PREVALENCE  \
0     California  2015–2016  Total population   All ages               8.7%   
1        Alameda  2015–2016  Total population   All ages               9.1%   
2         Alpine  2015–2016  Total population   All ages               9.3%   
3         Amador  2015–2016  Total population   All ages               9.3%   
4          Butte  2015–2016  Total population   All ages               9.4%   
...          ...        ...               ...        ...                ...   
1234      Tulare  2019–2020        Age groups  65+ years              13.3%   
1235    Tuolumne  2019–2020        Age groups  65+ years              10.3%   
1236     Ventura  2019–2020        Age groups  65+ years               7.5%   
1237        Yolo  2019–2020        Age groups  65+ years              14.0%   
1238        Yuba  2019–2020        Age groups  65+ years              14.4%   

     95% CONFIDENCE INTERVAL  \
0                  (8.1–9.3)   
1                 (5.9–12.2)   
2                 (4.1–14.6)   
3                 (4.1–14.6)   
4                 (3.8–15.1)   
...                      ...   
1234              (9.0–17.7)   
1235              (2.6–17.9)   
1236              (3.9–11.1)   
1237              (7.6–20.3)   
1238              (7.6–21.1)   

                                       COUNTIES GROUPED  \
0                                                   NaN   
1                                                   NaN   
2     Alpine, Amador, Calaveras, Inyo, Mariposa, Mon...   
3     Alpine, Amador, Calaveras, Inyo, Mariposa, Mon...   
4                                                   NaN   
...                                                 ...   
1234                                                NaN   
1235  Alpine, Amador, Calaveras, Inyo, Mariposa, Mon...   
1236                                                NaN   
1237                                                NaN   
1238                                                NaN   

                                                COMMENT  Unnamed: 8  \
0                                                   NaN         NaN   
1                                                   NaN         NaN   
2                                                   NaN         NaN   
3                                                   NaN         NaN   
4     Estimate is statistically unstable. Caution is...         NaN   
...                                                 ...         ...   
1234                                                NaN         NaN   
1235  Prevalence estimate is statistically unstable....         NaN   
1236                                                NaN         NaN   
1237                                                NaN         NaN   
1238                                                NaN         NaN   

      Unnamed: 9  Unnamed: 10  Unnamed: 11  Unnamed: 12  Unnamed: 13  \
0            NaN          NaN          NaN          NaN          NaN   
1            NaN          NaN          NaN          NaN          NaN   
2            NaN          NaN          NaN          NaN          NaN   
3            NaN          NaN          NaN          NaN          NaN   
4            NaN          NaN          NaN          NaN          NaN   
...          ...          ...          ...          ...          ...   
1234         NaN          NaN          NaN          NaN          NaN   
1235         NaN          NaN          NaN          NaN          NaN   
1236         NaN          NaN          NaN          NaN          NaN   
1237         NaN          NaN          NaN          NaN          NaN   
1238         NaN          NaN          NaN          NaN          NaN   

      Unnamed: 14  Unnamed: 15  
0             NaN          NaN  
1             NaN          NaN  
2             NaN          NaN  
3             NaN          NaN  
4             NaN          NaN  
...           ...          ...  
1234          NaN 

In [45]:
asthma.columns

Index(['COUNTY', 'YEARS', 'STRATA', 'AGE GROUP', 'CURRENT PREVALENCE',
       '95% CONFIDENCE INTERVAL', 'COUNTIES GROUPED', 'COMMENT', 'Unnamed: 8',
       'Unnamed: 9', 'Unnamed: 10', 'Unnamed: 11', 'Unnamed: 12',
       'Unnamed: 13', 'Unnamed: 14', 'Unnamed: 15'],
      dtype='object')

In [63]:
model = ols('[95% CONFIDENCE INTERVAL] ~ [COUNTY] * [YEARS]', data=asthma).fit()

model

SyntaxError: ignored

# **Regression**

In [42]:
# The dataset contains diabetes medical predictor variables. The patients in the dataset are women of Pima Native American heritage, who are 21 years old or older.
diabetes = pd.read_csv('https://raw.githubusercontent.com/amnasyed1/datasci_5_statistics/main/datasets/diabetes.csv')
diabetes

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0              6      148             72             35        0  33.6   
1              1       85             66             29        0  26.6   
2              8      183             64              0        0  23.3   
3              1       89             66             23       94  28.1   
4              0      137             40             35      168  43.1   
..           ...      ...            ...            ...      ...   ...   
763           10      101             76             48      180  32.9   
764            2      122             70             27        0  36.8   
765            5      121             72             23      112  26.2   
766            1      126             60              0        0  30.1   
767            1       93             70             31        0  30.4   

     DiabetesPedigreeFunction  Age  Outcome  
0                       0.627   50        1  
1                       0.351   31        0  
2                       0.672   32        1  
3                       0.167   21        0  
4                       2.288   33        1  
..                        ...  ...      ...  
763                     0.171   63        0  
764                     0.340   27        0  
765                     0.245   30        0  
766                     0.349   47        1  
767                     0.315   23        0  

[768 rows x 9 columns]

## Hypothesis

Hypothesis: There is no correlation with high BMI and glucose levels.
- BMI is less than 18.5 = underweight range
- BMI is 18.5 to <25 = healthy weight range
- BMI is 25.0 to <30 = overweight range.
- BMI is 30.0 or higher = obesity range.

Reference for indicating BMI ranges:
https://www.cdc.gov/obesity/basics/adult-defining.html#:~:text=If%20your%20BMI%20is%20less,falls%20within%20the%20obesity%20range.

- Null Hypothesis (H0): There is no correlation between high BMI and glucose levels

- Alternative Hypothesis (H1): There is a correlation between high BMI aand glucose levels